elide-datastore/elide-datastore-inmemorydb/src/main/java/com/yahoo/elide/datastores/inmemory/InMemoryTransaction.java

```diff
@@ -98,21 +100,16 @@ public void commit() {
     }
 
     @Override
-    public <T> T createObject(Class<T> entityClass) {
+    public void createObject(Object entity, RequestScope scope) {
+        Class entityClass = entity.getClass();
         if (dataStore.get(entityClass) == null) {
             dataStore.putIfAbsent(entityClass, new ConcurrentHashMap<>());
             TYPEIDS.putIfAbsent(entityClass, new AtomicLong(1));
         }
         AtomicLong idValue = TYPEIDS.get(entityClass);
         String id = String.valueOf(idValue.getAndIncrement());
-        try {
-            T instance = entityClass.newInstance();
-            setId(instance, id);
-            return instance;
-        } catch (InstantiationException | IllegalAccessException e) {
-            e.printStackTrace();
-        }
-        return null;
+        setId(entity, id);
+        operations.add(new Operation(id, entity, entity.getClass(), false));
```

Comment:
operations holds all changes made, and will execute each operation during commit. createObject() used to follow by save() which does this job, but now we don't call save() after it anymore (we create with populated object, and then commit directly). So we need to add this to operations.